### DISCUSSION TITLE
Vlang Has 3 Published Books Written About It (2023)

### DISCUSSION THREAD

**Discussion by Wajinn** [vlang/v#17458] [General]
It's a quite interesting to see that V has 3 books on it.  That there isn't a Wikipedia page, yet, is a bit odd, if not scandalous. 
Note - see updated book list ([25929](https://github.com/vlang/v/discussions/25929)).

[The V Programming Language basic](https://www.amazon.com/Programming-Language-basic-Japanese-ebook/dp/B08BKJDRFR) (Japanese), 2020
[Getting Started with V Programming](https://www.amazon.com/Getting-Started-Programming-end-end-ebook/dp/B09FKK3JL7), 2021
[Beginning with V Programming](https://www.amazon.com/Beginning-Programming-beginners-programming-language/dp/B09XSZKTJ1), 2022

**Comment by JalonSolov** [comment]
The page exists in Draft mode.  It wasn't accepted by Wikipedia before, as they needed more references, etc.

Getting Started with V Programming was already listed on the page.  I just added the other 2.

https://en.wikipedia.org/wiki/Draft:V_(programming_language)

**Reply by Wajinn** [reply]
Good looking draft.  Additionally, it appears they accept articles and blog posts that elaborate on features about the programming language.  So, will find 7 good links and post them here, sometime today.  They will arguably be about equal quality of those used for reference purposes by competing languages.  Might be good to have them, as another card to throw on the table.  But, as we know, the 3 published books should outright trump random articles and blog posts that were accepted as references for a number of other languages.

Note:  The references to 3rd parties might seem strange, but the often expected perspective, is that of a 3rd party who is writing about the subject and who is also using 3rd party sources of information to confirm statements.  3rd party sources are usually expected to be some sort of authority on the subject, able to confirm a statement in some way, or offers some kind of additional observable proof to the reader.  That any reference is acceptable, can be a matter of opinion, so may or may not be challenged.  Additionally, here's an interesting link to [EverybodyWiki](https://en.everybodywiki.com/index.php?title=V_(programming_language)), which demonstrates what is being referred to and provides a few more references.

Edit:  

Additional links have been ordered in a way, to be used for reference purposes (this is only suggestive):

First appeared June 22, 2019; 3 years ago[1]

Stable release	0.3.3[2]

designed by Alexander Medvednikov[3]

inspired by the Go programming language but also influenced by: C, Rust, and Oberon[4][5]

V is to be easy to use,[6][7]

variable shadowing is not allowed [8]

As fast as C (V's main backend compiles to human readable C), with equivalent code.[9][10][11]

V is written in V and compiles in less than a second.[12]

V can translate your entire C project[13][14]

Translating DOOM from C to V and building it takes less than a second.[15]

**References:**

1. https://github.com/vlang/v/releases?page=15

2. https://github.com/vlang/v/releases

3. Medvednikov, Alexander
https://github.com/medvednikov

4. James, Ben. "The V Programming Language: Vain Or Virtuous?" Hackaday.
https://hackaday.com/2019/07/23/the-v-programming-language-vain-or-virtuous/
Retrieved 23 July 2019.

5. Umoren, Samuel. "Building a Web Server using Vlang" Section.
https://www.section.io/engineering-education/building-web-server-with-vlang/
Retrieved 5 April 2021.

6. Jonah, Victor. "What is Vlang? An introduction" LogRocket.
https://blog.logrocket.com/what-is-vlang-an-introduction/
Retrieved 25 February 2021.

7. Knott, Simon. "An introduction to V"
https://simonknott.de/articles/vlang
Retrieved 27 June 2019.

8. "Introducing the V Tutorial!" Replit.
https://replit.com/talk/learn/Introducing-the-V-Tutorial/112856
Retrieved 4 January 2021.

9. discussions/7849 ("https://github.com/vlang/v/" removed from front to stop unnecessary link from here)

10. Shóstak, Vic. "The V programming language" DEV.
https://dev.to/koddr/good-to-know-the-v-programming-language-k5b
Retrieved 8 November 2021.

11. "C is how old now? - Learning the V programming language" l-m.
https://l-m.dev/cs/learning_v/
Retrieved 10 April 2022.

12. "V , the programming language that is making a lot of success" Terminal Root.
https://terminalroot.com/vlang-the-programming-language-that-is-making-a-lot-of-success/
Retrieved 18 January 2022.

13. https://github.com/vlang/c2v

14. https://github.com/vlang/v/blob/master/doc/docs.md#translating-c-to-v

15. https://youtu.be/6oXrz3oRoEg


Additional **Further Reading (under the 3 books):**

Tsoukalos, Mihalis. "Discover the V language" Linux Format 288.
https://issuu.com/futurepublishing/docs/lxf288.issuu/2
Published 29 March 2022.

Additional **External links** (that appear acceptable in comparison to other languages):

https://blog.vlang.io/ 

https://www.youtube.com/@VLang 

An introduction to V - the vlang
https://youtu.be/ZO3vlwqvTrc

**Comment by rraammiinn** [comment]
a youtube channel would be nicer .

**Reply by Wajinn** [reply]
Not understanding this comment.  Maybe the issue is not searching or try different search terms.  

V does have an official YouTube channel.  Also, there are various V users making videos about V as well (on different channels).

Go here- https://www.youtube.com/@VLang

**Reply by rraammiinn** [reply]
I saw it before but I also saw a channel for rust called let's get rusty that goes through all sections of rust book in a playlist and has other playlist for news and other sthings .
v and zig languages don't have such a comprehensive youtube channels .

**Reply by Wajinn** [reply]
Oh, a playlist of all (informative and instructional) videos about Vlang would be really cool.  Think that is a great idea.

**Comment by Wajinn** [comment]
> The page exists in Draft mode. https://en.wikipedia.org/wiki/Draft:V_(programming_language)

Just an opinion, but after reading the draft, think there are a few points that might be contentious.  This opinion about this, comes from reading things posted by detractors, so possibly modifying the below lines can be of some help.  This is only a suggestion, it is of course up to team V, as to what they want to put.

From [draft](https://en.wikipedia.org/wiki/Draft:V_(programming_language)):

> C interop without any costs

Change to- C interop

Reason- "...without any costs", can be a point of contention versus a statement of fact about what V can do.

> As fast as C

Change to- Can be as fast as C 

Reason- The difference is subtle, but can be quite significant in English.  "As fast", becomes another point of contention, where it can be purposely misinterpreted to be "as fast as C" in every circumstance and in any comparison.  "Can be as fast", is clarifying that the speeds seen can depend on various conditions.

> Autofree can be enabled with -autofree. It takes care of most objects (~90-100%): the compiler inserts necessary free calls automatically during compilation. Remaining small percentage of objects is freed via GC. The developer doesn't need to change anything in their code. "It just works", like in Python, Go, or Java, except there's no heavy GC tracing everything or expensive RC for each object. 

Change to- Autofree can be enabled with -autofree. It takes care of most objects (~90-100%): the compiler inserts necessary free calls automatically during compilation. Remaining small percentage of objects is freed via GC. 

Reasons:

> The developer doesn't need to change anything in their code. "It just works", like in Python, Go, or Java, except there's no heavy GC tracing everything or expensive RC for each object.  

Taken as a whole, creates multiple points of contention, and can be viewed as advertising.  

**A)** "The developer doesn't need to change anything...", traps team V into having to prove this and opens a vector of attack by detractors, versus simply a statement of fact about what autofree can do.  

**B)** "It just works", will debatably invite challenges and drama.  While a sales pitch is perfectly acceptable on one's website, Wikipedia overlords and detractors are bound to see something they can jump on.

**C)** "...except there's no heavy GC tracing everything or expensive RC for each object", appears to be contradictory to the previous statement, "objects is (are) freed via GC".  While it appears that the use of "no heavy" and "expensive" is attempting to show it's a matter of degree, it does invite confusion.  Therefore, it might be better to omit the other sentences, because the first few sentences are a statement of fact about how autofree works.

**Reply by medvednikov** [reply]
Can be as fast as C is not a great description. Python can be as fast as C.

V and C are on the same level in terms of speed. It is as fast as C, the benchmarks show that.

**Reply by medvednikov** [reply]
I like what it says on the home page now:
"As fast as C (V's main backend compiles to human readable C), with equivalent code.
V does introduce some overhead for safety (such as array bounds checking, GC free), but these features can be disabled/bypassed when performance is more important."

**Reply by JalonSolov** [reply]
... and that's what I pasted onto the Draft page.  :-)

**Reply by Wajinn** [reply]
> Can be as fast as C is not a great description. Python can be as fast as C.
> 
> V and C are on the same level in terms of speed. It is as fast as C, the benchmarks show that.

I agree with you, and understand what you mean.  However, I was coming from the view and context of detractors and Wikipedia overlords, who might want to challenge such statements (as it appears they are doing at the top of the draft page).  Therefore, it might be good to add reference links.  Then it can be seen as a statement with some validation or confirmation, versus an attempt at just advertising or merely a claim.

We might want it to look something like- As fast as C (V's main backend compiles to human readable C), with equivalent code.[11][12]
And [11][12] would have additional elaboration, proof, or substantiation in a reference section at the bottom of the page.
